AK/HashTable.h

```diff
@@ -70,19 +54,21 @@ class HashTableIterator {
             return;
         do {
             ++m_bucket;
-            if (m_bucket->state == BucketState::Used)
+            if (m_bucket == m_end_bucket) {
+                m_bucket = nullptr;
                 return;
-        } while (m_bucket->state != BucketState::End);
-        if (m_bucket->state == BucketState::End)
-            m_bucket = nullptr;
+            }
+        } while (m_bucket->state == BucketState::Free);
     }
 
-    explicit HashTableIterator(BucketType* bucket)
+    explicit HashTableIterator(BucketType* bucket, BucketType* end_bucket)
```

Comment:
`explicit` is no longer needed here

Reply:
Missed that, also applicable to the other iterator type. Removed both!